plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -221,14 +245,18 @@ func TestGetLatencies(t *testing.T) {
 				BaseQueryParameters: buildTestBaseQueryParametersFrom(tc),
 				Quantile:            0.95,
 			}
+			tracer, exp, closer := tracerProvider(t)
 			cfg := defaultConfig
 			if tc.updateConfig != nil {
 				cfg = tc.updateConfig(cfg)
 			}
-			reader, mockPrometheus := prepareMetricsReaderAndServer(t, cfg, tc.wantPromQlQuery, nil)
+			reader, mockPrometheus := prepareMetricsReaderAndServer(t, cfg, tc.wantPromQlQuery, nil, tracer)
 			defer mockPrometheus.Close()
 
 			m, err := reader.GetLatencies(context.Background(), &params)
+			assert.NotEmpty(t, exp.GetSpans(), "Spans recorded during the test.")
+			assert.Len(t, exp.GetSpans(), 1, "HTTP request was traced and span reported")
+			defer closer()
 			require.NoError(t, err)
```

Comment:
Too late

Reply:
sorry! why we got it too late?

Reply:
Defer close is always called as soon as object is created.